prime-router/src/main/kotlin/azure/SubmissionFunction.kt

```diff
@@ -82,68 +86,89 @@ class SubmissionFunction(
      * It does not assume the user belongs to a single Organization.  Rather, it uses
      * the organization in the URL path, after first confirming authorization to access that organization.
      */
-    @FunctionName("getOrgSubmissions")
-    fun getOrgSubmissions(
+    @FunctionName("getOrgSubmissionsList")
+    fun getOrgSubmissionsList(
         @HttpTrigger(
-            name = "getOrgSubmissions",
+            name = "getOrgSubmissionsList",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
             route = "waters/org/{organization}/submissions"
         ) request: HttpRequestMessage<String?>,
         @BindingName("organization") organization: String,
     ): HttpResponseMessage {
-        return oktaAuthentication.checkAccess(request, organization, true) {
-            try {
-                val (qSortOrder, qSortColumn, resultsAfterDate, resultsBeforeDate, pageSize, showFailed) =
-                    Parameters(request.queryParameters)
-                val sortOrder = try {
-                    SubmissionAccess.SortOrder.valueOf(qSortOrder)
-                } catch (e: IllegalArgumentException) {
-                    SubmissionAccess.SortOrder.DESC
-                }
+        try {
+            // Do authentication
+            val claims = AuthenticationStrategy.authenticate(request)
```

Comment:
Again, note the okta-specific call has been replaced with a general call to authenticate.